openmdao/core/_checks.py

```diff
@@ -82,9 +82,9 @@ def _check_types_match(src, tgt, to_prom_name):
     if stype == ttype:
         return
 
-    src_indices = tgt.get('src_indices')
-    if src_indices and len(src_indices) == 1 and ttype == float:
-        return
+    if 'src_indices' in tgt:
+        if len(tgt['src_indices']) == 1 and ttype == float:
+            return
 
```

Comment:
Is .get a bad idea for performance? I started doing it because I saw you doing it (instead of the if and then the get_item)